python/foglamp/services/core/api/service.py

```diff
@@ -95,72 +96,105 @@ def get_service_records():
             category_desc = plugin_config['plugin']['description']
             config_mgr = ConfigurationManager(storage)
             await config_mgr.create_category(category_name=name,
                                              category_description=category_desc,
                                              category_value=plugin_config,
                                              keep_original_items=True)
-        except ImportError as ex:
-            raise web.HTTPInternalServerError(reason='Plugin "{}" import problem from path "{}". {}'.format(plugin, plugin_module_path, str(ex)))
         except Exception as ex:
+            await revert_scheduled_processes(plugin)  # Revert scheduled_process entry
             raise web.HTTPInternalServerError(reason='Failed to create plugin configuration. {}'.format(str(ex)))
 
-        # Next add a schedule to run the new service at startup
-        schedule = StartUpSchedule()  # TODO: For North plugin also?
-        schedule.name = name
-        schedule.process_name = name
-        schedule.repeat = datetime.timedelta(0)
-        schedule.exclusive = True
-        schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
-
-        # Save schedule
-        await server.Server.scheduler.save_schedule(schedule, is_enabled)
-        schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        # If all successful then lastly add a schedule to run the new service at startup
+        try:
+            schedule = StartUpSchedule()
+            schedule.name = name
+            schedule.process_name = name
+            schedule.repeat = datetime.timedelta(0)
+            schedule.exclusive = True
+            schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
+
+            # Save schedule
+            await server.Server.scheduler.save_schedule(schedule, is_enabled)
+            schedule = await server.Server.scheduler.get_schedule_by_name(name)
+        except StorageServerError as ex:
+            await revert_configuration(name)  # Revert configuration entry
+            await revert_scheduled_processes(name)  # Revert scheduled_process entry
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(ex.error))
+        except Exception as ins_ex:
+            await revert_configuration(name)  # Revert configuration entry
+            await revert_scheduled_processes(name)  # Revert scheduled_process entry
+            raise web.HTTPInternalServerError(reason='Failed to created schedule. {}'.format(str(ins_ex)))
 
         return web.json_response({'name': name, 'id': str(schedule.schedule_id)})
 
     except ValueError as ex:
         raise web.HTTPNotFound(reason=str(ex))
 
+
+async def check_schedules(schedule_name):
+    storage = connect.get_storage_async()
```

Comment:
please do this once is caller def, and pass to this and `check_scheduled_processes` and `revert_scheduled_processes` as needed!